src/app/printer/pdf/pdf.printer.ts

```diff
@@ -59,10 +60,24 @@ export class PdfPrinter implements Printer {
             const usageSheetWidthOffset = (width - refWidth - usageWidth) / 2;
             const usageSheetHeightOffset = (height - entries.length * usageLineHeight) / 2;
             Array.from(entries).forEach(([k, v], idx) => {
+                let paletteEntry = getPaletteEntryByColorRef(project.paletteConfiguration.palettes, k);
+
                 doc.rect(usageSheetWidthOffset, usageLineHeight * idx + usageSheetHeightOffset, refWidth, usageLineHeight);
                 doc.rect(usageSheetWidthOffset + refWidth, usageLineHeight * idx + usageSheetHeightOffset, usageWidth, usageLineHeight);
+                doc.rect(usageSheetWidthOffset + refWidth + usageWidth, usageLineHeight * idx + usageSheetHeightOffset, usageWidth, usageLineHeight);
+                
+                doc.setFillColor(paletteEntry.color.r,paletteEntry.color.g,paletteEntry.color.b);
+                doc.rect(usageSheetWidthOffset + refWidth + usageWidth + usageWidth / 4, usageLineHeight * idx + usageSheetHeightOffset + 1, usageWidth / 2, usageLineHeight -2, 'F');
+
+                doc.setTextColor(0, 0, 0);
                 doc.text(usageSheetWidthOffset + cellPadding, usageLineHeight * idx + usageSheetHeightOffset + usageLineHeight / 2 + this.fontSizeToHeightMm(fontSize), k);
                 doc.text(usageSheetWidthOffset + refWidth + cellPadding, usageLineHeight * idx + usageSheetHeightOffset + usageLineHeight / 2 + this.fontSizeToHeightMm(fontSize), "" + v);
+                
+                var symbolWidth = doc.getStringUnitWidth(paletteEntry.symbol) * doc.internal.getFontSize() / doc.internal.scaleFactor;
+                if (paletteEntry.color.r + paletteEntry.color.g + paletteEntry.color.b < 128 * 3) {
```

Comment:
I believe the test is made in order to change the text color based on background. If yes, I made the same thing in the svg printer, it would be nice to have the exact same test, or even put this into a utility file.

[here](https://github.com/maxcleme/beadifier/blob/master/src/app/printer/svg/svg.printer.ts#L102)

Reply:
Or even enhance the PaletteEntry model to compute these at loading. But it could come at a later time in a PR focusing perf, would be more easier to review.

Reply:
Ahh thanks for the tip, I haven't had much chance to look into the image functions yet. I'd be reviewing those once I'm happy with how the PDF exporter works. I'll update it to move the test defined inside the SVG file into utils and use that here as it seems more accurate.

Reply:
Updated my approach here. Let me know if this works how you expected @maxcleme